vncviewer/DesktopWindow.cxx

```diff
@@ -322,6 +324,27 @@ void DesktopWindow::setCursor(int width, int height,
 }
 
 
+void DesktopWindow::setCursorPos(const rfb::Point& pos)
+{
+  if (!mouseGrabbed) {
+    // Do nothing if we do not have the mouse captured.
+    return;
+  }
+#if defined(WIN32)
+  SetCursorPos(pos.x, pos.y);
+#elif defined(__APPLE__)
+  CGPoint new_pos;
+  new_pos.x = pos.x;
+  new_pos.y = pos.y;
+  CGWarpMouseCursorPosition(new_pos);
+#else // Assume this is Xlib
+  Window rootwindow = DefaultRootWindow(fl_display);
+  XWarpPointer(fl_display, rootwindow, rootwindow, 0, 0, 0, 0,
+               pos.x, pos.y);
+#endif
```

Comment:
The coordinates here look wrong. The VNC screen might not be at coordinates 0,0, so everything here needs to be adjusted by an offset.

Reply:
haha, this is my first change of a client application, so i basically stumbled upon a solution that worked for me in this one case (translation: please suggest what offset should be added).

i kinda guess that the `viewport->x/y()` is what we want, but have no way of confirming since i don't know any situations other than the one i can test locally.

Reply:
I see two test scenarios:

 1. Viewport doesn't match window. Just disable "Resize remote session..." in the options and make sure the session is larger or smaller than your client window
 2. Client is on different monitors. You need two monitors for this, and disable "Enable full-screen mode over all monitors"

I would suspect you need both `viewport->x/y()` and `window()->x/y()` to properly get screen coordinates.

Reply:
both cases seemed to work with a combination of `x_root() + viewport->x()`. thanks!